datafusion/src/sql/planner.rs

```diff
@@ -1274,6 +1274,24 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 })
             }
 
+            SQLExpr::Trim { expr, trim_where } => {
+                let fun = match trim_where {
+                    Some((trim_where, _expr)) => {
```

Comment:
Filed https://github.com/apache/arrow-datafusion/issues/935 to track supporting this extended syntax